external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -218,7 +224,11 @@ public void nextTuple() {
             }
 
             if (poll()) {
-                setWaitingToEmit(pollKafkaBroker());
+                try {
+                    setWaitingToEmit(pollKafkaBroker());
+                } catch (Exception e) {
+                    LOG.error("Failed to poll from kafka.", e);
```

Comment:
We have encountered many exceptions: KafkaFetchException, LeaderNoFoundException, etc. These are caused by temporarily kafka partition migration. I don't think we should restart the process in these cases.